misc/install_tabnine.sh

```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env bash
+
+# Based on https://github.com/codota/TabNine/blob/master/dl_binaries.sh
+# Download latest TabNine binaries
+set -o errexit
+set -o pipefail
+set -x
+
+rm TabNine || true # remove old link
+
+version=${version:-$(curl -sS https://update.tabnine.com/version)}
+
+case $(uname -s) in
+"Darwin")
+    platform="apple-darwin"
+    ;;
+"Linux")
+    platform="unknown-linux-gnu"
+    ;;
+esac
+# platform="unknown-linux-gnu"
+triple="$(uname -m)-$platform"
+
+cd $(dirname $0)
+path=$version/$triple/TabNine
+if [[ -f "binaries/$path" ]] && [[ -e "binaries/TabNine_$(uname -s)" ]]; then
+    ln -sf $path "binaries/TabNine_$(uname -s)"
+    exit
+fi
+echo Downloading version $version
+curl https://update.tabnine.com/$path --create-dirs -o binaries/$path
+chmod +x binaries/$path
+
+ln -sf $path "binaries/TabNine_$(uname -s)"
```

Comment:
Sorry. I want to avoid adding the ability to install some binaries on nvim-compe.

So the `tabnine` source should check `tabnine` command in $PATH.

Reply:
agree, maybe we could add this script to the wiki if people want to know how to install it

Reply:
Well, this is your project after all.. However, I do believe having 'batteries included' is better. After all, users have the option to install the binary if they so chose when they load the plugin.

If we leave it the the user, its one more dependency for them to track manually. Its much easier when the plugin manager takes care of that.

Can you elaborate on you opposition to adding the install script?

Reply:
In some environments installing binary on plugin activation is not an option, for example on [nixos](https://nixos.org/) or other os with nix package manager maaging vim plugins. [Here](https://github.com/NixOS/nixpkgs/blob/master/pkgs/misc/vim-plugins/overrides.nix#L744) you can see how completion-tabnine has nothing to do with tabnine install from inside of the plugin, it just depends on the separate tabnine package.

Reply:
Well, the actual script was taken from aca/completion-tabnine, so im not sure I follow your logic.
Anyway, we can have a slightly different strategy of locating the tabnine binary:
1) try 'TabNine' from $PATH
2) if (1) fails, try the current approach (which would work if the user opted to call the install script)
3) if all fails, show an appropriate message.

Would this work?

Reply:
I've just noticed that you suggest just an optional tabnine install. That will work for nix because in nix we just don't execute such install scripts (they are considered non-pure and non-reproducible).

Reply:
For example, if we provide `github completion` for `COMMIT_EDITMSG`, Should we add `gh` command install scripts?
And In this case, should we suggest configuration as below?

```viml
Plug 'hrsh7th/nvim-compe', { 'do': './misc/install_tabnine.sh | ./misc/install_gh.sh' }
```

I think this is over responsibility. (And Sorry. I don't well shellscript so I can't maintenance it).

Reply:
I understand you not wanting to maintain it. Its a valid consideration.

However, gh is more common, and much easier to install. TabNine is not and, moreover (alas), propietary. Every other instance of a plugin using TabNine (completeion-nvim, deoplete, and others) do offer the install script. The only difference is that the plugin is not maintained in the main repo, but as a separate plug. 

Would you prefere I create a separate repo for compe-tabnine? 
If so, can you please explain how to register the source in compe?

Reply:
Thank you for your words.

Yes. If we should provide installation scripts, the source should be separated repository (I think `compe-tabnine` is good).

The source registration is super simple and easy. You can create below structure.

```sh
/compe-tabnine
  /after
    /plugin
      compe_tabnine.vim
  /lua
    /compe_tabnine
      init.lua
```

Reply:
ok, I will switch to a seperate repo.
would appreciate if you can reference it in you documentation.